orthoinference/Jenkinsfile

```diff
@@ -1,214 +1,79 @@
 import groovy.json.JsonSlurper
-
+// This Jenkinsfile is used by Jenkins to run the Orthoinference step of Reactome's release.
+// It requires that the Orthopairs and UpdateStableIdentifiers steps have been run successfully before it can be run.
 pipeline{
 	agent any
-	
+
 	stages{
-		stage('Check if upstream builds succeeded'){
+		// This stage checks that upstream projects Orthopairs and UpdateStableIdentifier, were run successfully for their last build.
+		stage('Check if Orthopairs and UpdateStableIdentifiers builds succeeded'){
 			steps{
 				script{
-					def orthopairsStatusUrl = httpRequest authentication: 'jenkinsKey', url: "${env.JENKINS_JOB_URL}/job/${env.RELEASE_NUMBER}/job/Orthopairs/lastBuild/api/json"
-					def orthopairsStatusJson = new JsonSlurper().parseText(orthopairsStatusUrl.getContent())
-					if(orthopairsStatusJson['result'] != "SUCCESS"){
-						error("Most recent Orthopairs build status: " + orthopairsStatusJson['result'])
-					}
-			    	}
-				script{
-					def updateStIdsStatusUrl = httpRequest authentication: 'jenkinsKey', url: "${env.JENKINS_JOB_URL}/job/${env.RELEASE_NUMBER}/job/UpdateStableIdentifiers/lastBuild/api/json"
-					def updateStIdsStatusJson = new JsonSlurper().parseText(updateStIdsStatusUrl.getContent())
-					if(updateStIdsStatusJson['result'] != "SUCCESS"){
-						error("Most recent UpdateStableIdentifiers build status: " + updateStIdsStatusJson['result'])
-					}
-				}	
-		    	}
-	    	}
+					// This queries the Jenkins API to confirm that the most recent builds of Orthopairs and UpdateStableIdentifiers were successful.
+					checkUpstreamBuildsSucceeded("Orthopairs")
+					checkUpstreamBuildsSucceeded("UpdateStableIdentifiers")
+				}
+			}
+		}
+		// This stage backs up the release current database before it is modified.
 		stage('Setup: Backup release_current'){
 			steps{
 				script{
 					dir('orthoinference'){
 						withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
-							sh "mysqldump -u$user -p$pass ${env.RELEASE_CURRENT} > ${env.RELEASE_CURRENT}_${env.RELEASE_NUMBER}_before_orthoinference.dump"
-							sh "gzip -f ${env.RELEASE_CURRENT}_${env.RELEASE_NUMBER}_before_orthoinference.dump"
+							def release_current_before_orthoinference_dump = "${env.RELEASE_CURRENT}_${env.RELEASE_NUMBER}_before_orthoinference.dump"
+							sh "mysqldump -u$user -p$pass ${env.RELEASE_CURRENT} > $release_current_before_orthoinference_dump"
+							sh "gzip -f $release_current_before_orthoinference_dump"
 						}
 					}
 				}
 			}
-		}			
+		}
+		// This stage builds the jar file using maven. It also runs the Main orthoinference process as a 'sub-stage'.
+		// This was due to a restriction in iterating over a list of species names. To iterate, you need to first have a 'stage > steps > script hierarchy.
```

Comment:
Being very nitpicky :), is there a missing single closing quote?

Reply:
_SUUUUPER_ nitpicky! ;)

Reply:
You're good -- I bet you've knocked off a Where's Waldo or two in your day